proposals/NNNN-sdl-js-pwa.md

```diff
@@ -0,0 +1,213 @@
+
+> Example of a local web app presenting the user interface with the HTML renderer of the WebEngine.
+
+Widgets are still available and can be controlled using `Show` RPC. Any overlay like Alert, ChoiceSets, Slider etc. are also available to the application.
+
+##### 2.2.1 Mobile and HMI API
+
+```xml
+<enum name="AppHMIType" since="2.0">
+  :
+  <element name="OPEN_HMI" since="5.x">
+</enum>
+```
+
+##### 2.2.2 Policy control
+
+Using the new HMI type would be policy controlled. On the policy server this new HMI type can be added to the valid HMI type list per app ID. Only apps with permissions to use this new HMI type would be allowed to register.
+
+##### 2.2.3 Apps using Open HMI
+
+With the current depth of this chapter the new HMI type should be used by 1st party OEM apps only. With future proposals and workshops the SDLC should open the new HMI type to 3rd party by creating and defining proper driver distraction and user interface guidelines. 
+
+### Chapter 3: Web application package
+---------
+
+Running apps on an embedded WebEngine defines a new app platform/runtime. With installing an app on the infotainment system there should be a minimum set of requirements on how the app should be packaged.
+
+1. The app package should be a compressed zip file
+2. It should contain a manifest.json and the sdl.js file
+3. The manifest file should contain:
+   1. a relative path to an html file which is the entry point of the app
+   2. a relative path to an app icon in the app bundle
+   3. SDL app ID
+   4. app name
+   5. optionally per supporting SDL locale:
+      1. app name (overrides global app name) (must be one of the valid app nicknames)
+      2. optionally relative path to an app icon (overrides global app icon)
+      3. optionally TTS name (an array of tts chunks)
+      4. optionally VR app names (an array of strings)
+   6. App version
+   7. Min SDL version supported
+4. All script files should be included in the package
+   1. Any `<script>` element with `src` attribute should point to a script file in the package
+   2. No scripts from outside the package should be allowed
+5. The entry point HTML file should refer to the manifest file (`<script src="manifest.json" />`)
+
+Above metadata may be used for the OEM app store and for presenting the app on the HMI (including voice capability).
+
+This definition should ensure that apps can be approved and verified by SDLC and OEMs without possibility of modifications after approval. Also this set of requirements should ensure compatibility throughout integrators. The final approval process will be part of another proposal.
+
+## Potential downsides
+
+The upside of apps running with a WebEngine is that it comes with an extremely flexible html based user interface and with a very sandboxed runtime environment. This can also be seen as downsides as the responsibility of following driver distraction rules increases and that a sandbox environment is limiting the functionality of apps.
```

Comment:
```suggestion
The upside of apps running with a WebEngine is that it comes with an extremely flexible html based user interface and with a very sandboxed runtime environment. This can also be seen as a downside as the responsibility of following driver distraction rules increases and a sandbox environment is limiting the functionality of apps.
```